### DISCUSSION TITLE
Cannot select proper device type for a tuya heater after having removed it once

### DISCUSSION THREAD

**Discussion by elboletaire** [make-all/tuya-local#1544] [Q&A] [answered]
So I've recently added three heaters using tuya local, everything was working smooth, but I made a mistake and removed one of the heaters from tuya local.

Now, when trying to add it again, it does not show me the type `arlec_panel_heater_v2`, like it did (and is selected for the other heaters), and I cannot find a way to work around this... I already tried removing it from tuya, so it would generate a new local key, but that did nothing, and all I see in the type list are devices that are not meant to control a heater (basically, none of them have a thermostat).

![imatge](https://github.com/make-all/tuya-local/assets/153305/b2de6b18-3f2f-4839-973e-0724a5ee481b)

Is there any way I could add this manually somehow to the settings? I'm trying to find out how is it stored, so I could manually add it copy-pasting the config from the other two heaters (changing values ofc..), but I've not been able to find this data location.

Edit: It's actually being detected as `gold_cct_lightbulb` (?):
~~~log
2024-01-19 22:01:44.182 WARNING (MainThread) [custom_components.tuya_local.config_flow] Device matches gold_cct_lightbulb with quality of 57%. DPS: {"updated_at": 1705698098.8263006, "1": false, "2": 20, "3": 15, "7": false, "13": 0, "101": false, "102": -1}
2024-01-19 22:01:44.183 WARNING (MainThread) [custom_components.tuya_local.config_flow] Report this to https://github.com/make-all/tuya-local/issues/
2024-01-19 22:11:26.556 WARNING (MainThread) [custom_components.tuya_local.config_flow] Device matches gold_cct_lightbulb with quality of 57%. DPS: {"updated_at": 1705698681.9198418, "1": false, "2": 20, "3": 15, "7": false, "13": 0, "101": false, "102": -1}
2024-01-19 22:11:26.556 WARNING (MainThread) [custom_components.tuya_local.config_flow] Report this to https://github.com/make-all/tuya-local/issues/
~~~

**Comment by elboletaire** [comment]
> **Moderation note**: the below is bad advice and can lead to messing up your HA installation. If your device is not matching a config, then it is not compatible with that config, and the root of that needs to be fixed. In the OP's case, their device was not reporting the preset mode when it had been previously, the proper fix was to mark the preset_mode as optional in the config so it would match that case.

Found a way to, at least, add it and make it work again:
- Add it selecting the recommended device (in my case, `gold_cct_lightbulb`)
- Go to the homeassistant `.storage` folder and open `core.config_entries` file
- At the end of the file should appear the recently added tuya device, there:
- Change `data.type` to `arlec_panel_heater_v2` for the desired device and save the file
- Restart home assistant

Still, I guess there's some weird bug/issue, since adding it the very first time worked, and then there was no way to do so.

**Reply by elboletaire** [reply]
I was actually asking how to add it manually, so I'm marking my response as valid. If you're having issues with your device not being detected properly (ever), this is not the place for it.

I may consider opening an issue for this missdetection, since it doesn't make any sense (and I fear I could have to do the same if I ever do any changes to the heaters [which I'd avoid ofc]); I recommend you to do the same if you're having issues with your devices.

**Reply by BakersHub** [reply]
This worked for me thank you !!

After working out shh and showing hidden files i could finally access .storage

**Reply by elboletaire** [reply]
You made me realize I said I was marking this as the valid answer but never did so 😅

**Comment by make-all** [comment, marked as answer]
The preset mode is not being reported by the device. You may have more luck adding it while it is on.
I don't recommend adding a different device type then hacking the config, as it will leave a mess in the entity registry.

**Reply by elboletaire** [reply]
But that's weird.. why is not reporting it properly now? I even tried resetting the device and repairing it with tuya, so what could be causing it to behave different after having removed it from tuyalocal once?

**Reply by memattyg** [reply]
I am having a similar issue with the GoSund dimmer switch that you recently added support for (marked request as complete).  I can not get the device type to populate when I add the device whether it is on or off.

**Reply by Mannokia** [reply]
I have a similar problem with my thermostat. I had a device that worked, after some update the temperature control stopped working, I deleted the device and cannot add it again because there are no devices with the required type in the proposed list.

**Reply by make-all** [reply]
The logs might shed some light

**Reply by contactnat** [reply]
Same issue here with goldair geco heater, it worked fine until I deleted it, now there are no heaters in the drop down. @make-all  what do you think mate?
![Screenshot_20240506_213555_Home Assistant.jpg](https://github.com/make-all/tuya-local/assets/81800839/7102c1c8-fdd5-4dff-acd8-1aaea65a866a)



 Logs say:

Logger: custom_components.tuya_local.config_flow
Source: custom_components/tuya_local/config_flow.py:97
integration: Tuya Local (documentation, issues)
First occurred: 9:35:37 PM (1 occurrences)
Last logged: 9:35:37 PM

Device matches wfws02rj_dual_switch with quality of 60%. DPS: {"updated_at": 1714988136.8578806, "1": true, "2": false, "3": 37, "4": 37, "5": 0}

**Reply by contactnat** [reply]
@make-all  you can ignore my query above, I dont think the device is supported, so I have submitted a new device request with all the info required in the form. thanks for all your work on this, its really appreciated.

**Comment by make-all** [comment]
> I am having a similar issue with the GoSund dimmer switch that you recently added support for (marked request as complete). I can not get the device type to populate when I add the device whether it is on or off.

This is not a similar issue, as that is not a supported device in the release you are using, unless you specifically updated to the development version with that recent change.

**Comment by memattyg** [comment]
Sorry, I thought when the request was marked as complete that meant the device was added and now supported. What do I need to provide to get the device to be supported? Sorry if this is the wrong place, just a little confused now on how to get my device supported and recognized so I can use it with the integration. On Jan 25, 2024, at 9:46 AM, Jason Rumney ***@***.***> wrote:﻿

I am having a similar issue with the GoSund dimmer switch that you recently added support for (marked request as complete). I can not get the device type to populate when I add the device whether it is on or off.

This is not a similar issue, as that is not a supported device.

**Reply by elboletaire** [reply]
You should create a new issue, this is the discussions section of the project, not the issues one: https://github.com/make-all/tuya-local/issues

**Reply by make-all** [reply]
Marked as complete just means the work has been done. You need to wait for a release after that, at which time the issue will be closed.

**Comment by BakersHub** [comment]
Have the same issue with a Chinese presence sensor
Connected once was great for 3 hours, then it randomly dropped off the network and when reconnecting it, it believes its now a "loonas curtain"

Tried reinstalling everything
Removing and re adding device on tuya
Removed 
    Tuya local
    Full reinstall of Home assistant

Feel like I'm out of rollback options

Anyone find a solution to this?
Or is it stuck being linked to the cloud, with no local support?

**Comment by ne0c0de** [comment]
same here like @BakersHub issue, I bought a chinese human presence sensor, it works almost well in Tuya but when I added it to HA via official Tuya integration, there were no entities.

Then I tried Tuya Local (not Local Tuya) and try to add device via SmartLife cloud-assisted device setup.

It found all the required device information but on the next step it asks for device type but there's no option for human presence sensor neither any sensor.

Then I selected any other option but after completing the steps, it's not show any sensor entity in it.

So the question is: How can I change the device type after I adding it?

**Reply by make-all** [reply]
No, you can't change the device type after adding, and doing so will not magically make your device work if it is unsupported.

**Comment by tashgitt** [comment]
I have such an experience with Frizzlife LP365P water monitor. I had it installed just fine a month or two ago. There was a power outage and the water monitor was coming back with "Failed to setup will retry..." I deleted and tried to add it again, but now I am not given a choice for the correct device "Frizzlife LP365P". Instead I am given some generic devices:
<img width="768" height="619" alt="Tuya local add smart water monitor 2" src="https://github.com/user-attachments/assets/456bf730-be97-4436-b410-14943192ff34" />

**Reply by tashgitt** [reply]
Well, i tried entering pairing mode multiple times, added it to SmartLife over and over again, with no result. Still not showing up correctly at TUYA LOCAL. Finally decided to do a factory reset and it worked.

**Comment by chrisequalsdev** [comment]
I ran into a similar issue adding some smart light bulbs; there were no suitable device options. I resolved it by temporary permitting WAN access again for the device IP's then re-adding them home assistant. Seems to still need WAN access during initial setup. Would be something to check if you are using a firewall to block internet access.
